src/main/thrift/parquet.thrift

```diff
@@ -595,6 +607,43 @@ union ColumnOrder {
   1: TypeDefinedOrder TYPE_ORDER;
 }
 
+struct PageLocation {
+/** Offset of the page in the file **/
+  1: required i64 offset
+
+/** Size of the page, including header. The same as PageHeader.compressed_page_size **/
+  2: required i32 compressed_page_size
+
+/** Index within the RowGroup of the first row of the page **/
+  3: required i64 first_row_index
+}
+
+struct OffsetIndex {
+/** PageLocations, ordered by increasing PageLocation.offset **/
+  1: required list<PageLocation> page_locations
+}
+
+/**
+ * Description for ColumnIndex.
+ * Each <array-field>[i] refers to the page at OffsetIndex.page_locations[i]
+ */
+struct ColumnIndex {
+/** A list of bools to determine the validity of the corresponding min and max values **/
+  1: required list<bool> valid_values
+
+/** A list containing the lower bounds for the values of each page **/
+  2: required list<binary> min_values
+
+/** A list containing the upper bounds for the values of each page **/
+  3: optional list<binary> max_values
+
+/** A list containing DataPageHeaderV2.statistics.null_count for each page **/
+  4: optional list<i64> null_count
+
+/** A list containing DataPageHeaderV2.statistics.distinct_count for each page **/
+  5: optional list<i64> distinct_count
```

Comment:
I think we agreed to remove this one in the last sync?

Reply:
Actually, I think we need this because this structure is replacing `Statistics`, which can store a distinct count. I don't know that anyone actually writes distinct count (Parquet MR doesn't) but we should make sure before we remove it entirely.

Reply:
Impala also doesn't populate the field. We could keep it out in this change and check other readers (which ones)? Once we drop the page statistics, we can either add it here as an optional, or not.